echopype/echodata/echodata.py

```diff
@@ -312,6 +312,145 @@ def squeeze_non_scalar(n):
                 "this method only supports sonar_model values of AZFP, EK60, and EK80"
             )
 
+    def update_platform(self, extra_platform_data: xr.Dataset, time_dim="time"):
+        """
+        Updates the `EchoData.platform` group with additional external platform data.
+
+        `extra_platform_data` must be an xarray Dataset.
+        The name of the time dimension in `extra_platform_data` is specified by the
+        `time_dim` parameter.
+        Data is extracted from `extra_platform_data` by variable name; only the data
+        in `extra_platform_data` with the following variable names will be used:
+            - `"pitch"`
+            - `"roll"`
+            - `"heave"`
+            - `"latitude"`
+            - `"longitude"`
+            - `"water_level"`
+        The data inserted into the Platform group will be indexed by a dimension named `"time2"`.
+
+        Parameters
+        ----------
+        extra_platform_data : xr.Dataset
+            An `xr.Dataset` containing the additional platform data to be added
+            to the `EchoData.platform` group.
+        time_dim: str, default="time"
+            The name of the time dimension in `extra_platform_data`; used for extracting
+            data from `extra_platform_data`.
+
+        Examples
+        --------
+        >>> ed = echopype.open_raw(raw_file, "EK60")
+        >>> extra_platform_data = xr.open_dataset(extra_platform_data_file)
+        >>> ed.update_platform(extra_platform_data)
+        """
+
+        # # only take data during ping times
+        # start_time, end_time = min(self.beam["ping_time"]), max(self.beam["ping_time"])
+
+        # Handle data stored as a CF Trajectory Discrete Sampling Geometry
+        # http://cfconventions.org/Data/cf-conventions/cf-conventions-1.8/cf-conventions.html#trajectory-data
+        # The Saildrone sample data file follows this convention
```

Comment:
@emiliom: reading this section you pointed out makes me wonder whether or not it is worthwhile to point out that the lat/lon data in the SONAR-netCDF4 ver.1 is consistent with the H.4.2. Single trajectory specification. Not that this necessitates any actual changes, but there seems to be values to make this explicit.

Reply:
Also, I think once we get around to put this function into the documentation, we should link this explicitly!

Reply:
> @emiliom: reading this section you pointed out makes me wonder whether or not it is worthwhile to point out that the lat/lon data in the SONAR-netCDF4 ver.1 is consistent with the H.4.2. Single trajectory specification. Not that this necessitates any actual changes, but there seems to be values to make this explicit.

Late reply ... Interesting observation. I agree, the structure is consistent. Maybe the right place to add this observation is the SONAR-netCDF4 version 2 draft text. From what Gavin said, it doesn't look like there are changes in the Platform group in v.2. 

> Also, I think once we get around to put this function into the documentation, we should link this explicitly!

Definitely.

Reply:
The consistency with the CF single trajectory specification was deliberate :) We tried to use CF conventions whenever it was sensible (and did attempt to meet CF-1.7, although it wasn't feasible for all of sonar-netcdf4 v1).

Reply:
@gavinmacaulay thanks for taking the time to chime in! I've gone ahead and created a new issue in the SONAR-netCDF4 repo. I'll try to handle it myself, hopefully next week.